Source Code/Ticket Management System/customer_menu.c

```diff
@@ -105,8 +110,38 @@ void viewFAQ()
 }
 
 void updateContactInfo()
-{
-	// TODO
+{	
+	system("CLS");
+	char email[512];
+	wchar_t password[50];
+	wchar_t name[50];
+	wchar_t phone_number[20];	
+	struct User* logged_user=auth_getUser(); //fetchs the user from the container
+	wprintf(L"Contact Info update\n");
+	wprintf(L"===============\n");
+	wprintf(L"enter 'no' every field you wish to not update.\n");
+	wprintf(L"Update email:\n");
+	input_char(email, 512);
+	wprintf(L"Update password:\n");
+	input_wchar(password, 50);
+	wprintf(L"Update name:\n");
+	input_wchar(name, 50);
+	wprintf(L"Update Phone number:.\n");
+	input_wchar(phone_number, 50);
+	if (!input_valid_email(email))
```

Comment:
Maybe first check if it's `"no"` and then if it's valid?
`"no"` is not going to be a valid email.

Reply:
you are right, switched it around